### PR TITLE
hyperdrive outgoing

### PR DESCRIPTION
**Pull request by gitsimon**

### CODE REVIEW

File: pallets/hyperdrive-outgoing/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+                );
+                MessageNumbers::<T, I>::put((next_message_number, next_message_number));
+            }
+
+            // TODO add weight used here in on_initialize (convention)
+        }
+
+        fn on_initialize(_n: T::BlockNumber) -> Weight {
+            T::WeightInfo::on_initialize()
+        }
```
Comment (godenzim) [inline pallets/hyperdrive-outgoing/src/lib.rs:234]: @gitsimon I do not get why we need this

File: pallets/hyperdrive-outgoing/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+
+serde = { version = "1.0.136", features = ["derive"], optional = true }
+thiserror = "1.0"
+log = { version = "0.4.17", default-features = false }
+
+[dev-dependencies]
+hex-literal = "0.3"
+
+sp-io = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36", default-features = false }
```
Comment (RomarQ) [inline pallets/hyperdrive-outgoing/Cargo.toml:59]:
```suggestion
```
Duplicated dependency, is already set as non dev dependency

File: pallets/hyperdrive-outgoing/src/types.rs [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+use sp_runtime::traits;
+#[cfg(not(feature = "std"))]
+use sp_std::prelude::Vec;
+use sp_std::prelude::*;
+
+use pallet_acurast::{JobIdSequence, TezosAddressBytes};
+use strum_macros::{EnumString, IntoStaticStr};
+
+/// Prefix for elements stored in the Off-chain DB via Indexing API.
+pub const INDEXING_PREFIX: &'static [u8] = b"ammr";
```
Comment (RomarQ) [inline pallets/hyperdrive-outgoing/src/types.rs:18]: The prefix should probably contain the target chain identifier.

**Review by RomarQ — APPROVED** [review]

File: pallets/hyperdrive-outgoing/Cargo.toml (hunk 2 above)
Comment (gitsimon) [reply pallets/hyperdrive-outgoing/Cargo.toml:59]: removed

File: pallets/hyperdrive-outgoing/src/types.rs (hunk 3 above)
Comment (gitsimon) [reply pallets/hyperdrive-outgoing/src/types.rs:18]: that was a left-over, the prefix is specified in the config, specific for each instance=target chain

**Review by RomarQ — APPROVED** [review]

**Review by godenzim — APPROVED** [review]

**Pull request merged**